src/vsr/replica.zig

```diff
@@ -9936,13 +9945,15 @@ pub fn ReplicaType(
                 return;
             }
 
-            if (self.journal.writing(message.header.op, message.header.checksum)) {
-                log.debug("{}: write_prepare: ignoring op={} checksum={} (already writing)", .{
-                    self.replica,
-                    message.header.op,
-                    message.header.checksum,
-                });
-                return;
+            switch (self.journal.writing(message.header)) {
+                .none => {},
+                .slot, .exact => |reason| {
+                    log.debug(
+                        "{}: write_prepare: ignoring op={} checksum={} (already writing {})",
+                        .{ self.replica, message.header.op, message.header.checksum, reason },
+                    );
```

Comment:
```suggestion
                    log.debug(
                        "{}: write_prepare: ignoring op={} checksum={} (already writing {s})",
                        .{ self.replica, message.header.op, message.header.checksum, @tagName(reason) },
                    );
```

I _think_ just printing enum would print `Reason.writing`?

Reply:
Updated. I knew that but didn't want to go over 100 characters... :stuck_out_tongue: